File: PHP/MinimumPathSum.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

class MinimumPathSum {    

    /**
     * @param Integer[][] $grid
     * @return Integer
     */
    function minPathSum($grid) {
        $m = count($grid);
        $n = count($grid[0]);

        for ($i=1; $i < $m; $i++) $grid[$i][0] += $grid[$i-1][0];
        for ($j=1; $j < $n; $j++) $grid[0][$j] += $grid[0][$j-1];
        for ($i=1; $i < $m; $i++) {
            for ($j=1; $j < $n; $j++) $grid[$i][$j] += min($grid[$i-1][$j],$grid[$i][$j-1]);
        }

        return $grid[$m-1][$n-1];
    }
}

File: GO/MinimumPathSum.go
/**
 * @version GO 1.20.2
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

func minPathSum(grid [][]int) int {
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[0]); j++ {
			if i == 0 && j > 0 {
				grid[i][j] = grid[i][j-1] + grid[i][j]
			} else if i > 0 && j == 0 {
				grid[i][j] = grid[i-1][j] + grid[i][j]
			} else if i > 0 && j > 0 {
				grid[i][j] = min(grid[i-1][j], grid[i][j-1]) + grid[i][j]
			}
		}
	}
	return grid[len(grid)-1][len(grid[0])-1]
}

func min(a int, b int) int {
	if a > b {
		return b
	} else {
		return a
	}
}
